app/models/contribution.rb

```diff
@@ -45,29 +45,15 @@ def self.author_valid?(contrib)
     end
   end
 
-  def self.valid_fields?(contrib)
-    featured_valid?(contrib) &&
-      status_valid?(contrib) &&
-      visibility_valid?(contrib)
-  end
-
-  # @return [Boolean]
-  def self.visibility_valid?(contrib)
-    contrib = contrib.with_indifferent_access
-    VISIBILITY_VALUES.include? contrib[:visibility].to_s.downcase
-  end
-
+  # checks featured, status and visibility
   # @return [Boolean]
-  def self.status_valid?(contrib)
-    contrib = contrib.with_indifferent_access
-    STATUS_VALUES.include? contrib[:status].to_s.downcase
-  end
-
-  # Allowed values for featured are true and false
-  # @return [Boolean]
-  def self.featured_valid?(contrib)
-    contrib = contrib.with_indifferent_access
-    contrib[:featured].to_s =~ /true|false/i ? true : false
+  def self.valid_fields?(contrib)
+    segment = contrib.with_indifferent_access.slice(:featured, :status, :visibility)
+    return false unless segment.size == 3
+    return false if segment.values.any?(&:nil?)
+    prototype = Contribution.new(segment)
+    prototype.validate # we KNOW it won't validate (w/o author and publication), but we check for the other fields
+    prototype.errors.messages.slice(:featured, :status, :visibility).empty?
   end
```

Comment:
👌 - only wondering why all the prelude checks instead of just throwing `contrib` to `Contribution.new(contrib)`.  As our own code is likely the only consumer of this method, it's probably going to give it the right stuff.  The only other thought is what happens to an API PATCH that only needs to validate one field?

Reply:
I agree that is the goal.  This moves us in that direction.  I implement more in that direction in other PRs.

Reply:
:ok_hand: